### PR TITLE
Used same immutable event data for all listeners

### PR DESCRIPTION
**Pull request by ahmetmircik**
Used same immutable event data for all listeners

### CODE REVIEW

**Comment by ahmetmircik** [comment]
Failed test:
`ClientReliableTopicOverloadTest.whenBlock_whenNoSpace`: https://hazelcast-l337.ci.cloudbees.com/job/new-lab-fast-pr/6303/

**Comment by ahmetmircik** [comment]
run-lab-run

File: hazelcast/src/main/java/com/hazelcast/internal/nearcache/impl/invalidation/BatchInvalidator.java [hunk 1]
```diff
@@ -133,10 +133,11 @@ private void createAndSendInvalidations(String mapName, InvalidationQueue invali
     }
 
     private void sendInvalidations(String mapName, List<Invalidation> invalidations) {
+        Invalidation invalidation = new BatchNearCacheInvalidation(mapName, invalidations);
```
Comment (pveentjer) [inline hazelcast/src/main/java/com/hazelcast/internal/nearcache/impl/invalidation/BatchInvalidator.java]: Good work.

One thing that could be a potential problem (I don't enough about how this is being used), but in the old code, if there are no listener, no BatchNearCacheInvalidation is created, unlike the new code. One way to fix this is to create the BatchNearCacheInvalidation lazily.

Comment (ahmetmircik) [inline hazelcast/src/main/java/com/hazelcast/internal/nearcache/impl/invalidation/BatchInvalidator.java]: yes there will always be one instance.

Comment (pveentjer) [inline hazelcast/src/main/java/com/hazelcast/internal/nearcache/impl/invalidation/BatchInvalidator.java]: Then I would say: good work and :+1: 

Can you add a line of comment explaining the assumption? For future generations..

Comment (ahmetmircik) [inline hazelcast/src/main/java/com/hazelcast/internal/nearcache/impl/invalidation/BatchInvalidator.java]: comment added here: https://github.com/hazelcast/hazelcast/commit/15a3ad94d2603f37d05342941d963f993b4cec2e

**Pull request merged**